lib/live_debugger/app/debugger/streams/stream_utils.ex

```diff
@@ -0,0 +1,244 @@
+defmodule LiveDebugger.App.Debugger.Streams.StreamUtils do
+  @moduledoc """
+  Utilities for extracting Phoenix.LiveView.Stream diffs
+  from render traces and mapping them into a list of functions.
+  """
+
+  @type live_stream_item :: %Phoenix.LiveView.LiveStream{
+          name: atom(),
+          dom_id: (any() -> String.t()),
+          ref: String.t(),
+          inserts: list(),
+          deletes: list(),
+          reset?: boolean(),
+          consumable?: boolean()
+        }
+
+  @type stream_entry :: %{
+          optional(atom()) => live_stream_item(),
+          __changed__: MapSet.t(atom()),
+          __configured__: %{optional(atom()) => [dom_id: (any() -> String.t())]},
+          __ref__: any()
+        }
+
+  @spec extract_stream_traces([LiveDebugger.Structs.Trace.FunctionTrace.t()]) ::
+          [stream_entry()]
+  def extract_stream_traces(stream_updates) do
+    stream_updates =
+      stream_updates
+      |> Enum.sort_by(& &1.timestamp, :asc)
+      |> Enum.flat_map(& &1.args)
+      |> Enum.map(&Map.get(&1, :streams, []))
+      |> Enum.reject(&Enum.empty?/1)
+
+    stream_updates
```

Comment:
You don't have to create a new variable if you immediately return it 😄 

```suggestion
    stream_updates
    |> Enum.sort_by(& &1.timestamp, :asc)
    |> Enum.flat_map(& &1.args)
    |> Enum.map(&Map.get(&1, :streams, []))
    |> Enum.reject(&Enum.empty?/1)
```